### PR TITLE
Updating Event Grid  package reference

### PR DESCRIPTION
**Pull request by fabiocav**
Updating Event Grid extension reference to 3.2.1

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: extensions/Worker.Extensions.EventGrid/src/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-﻿using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
+using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
 
-[assembly: ExtensionInformation("Microsoft.Azure.WebJobs.Extensions.EventGrid", "3.2.0")]
+[assembly: ExtensionInformation("Microsoft.Azure.WebJobs.Extensions.EventGrid", "3.2.1")]
```
Comment (kshyju) [inline extensions/Worker.Extensions.EventGrid/src/Properties/AssemblyInfo.cs:6]: Don't we need to update the .csproj as well? 

https://github.com/Azure/azure-functions-dotnet-worker/blob/main/extensions/Worker.Extensions.EventGrid/src/Worker.Extensions.EventGrid.csproj#L9

Comment (fabiocav) [reply extensions/Worker.Extensions.EventGrid/src/Properties/AssemblyInfo.cs:6]: Yes, that change wasn't added to my commit. Fixed.

**Review by kshyju — APPROVED** [review]

**Pull request merged**